packages/@uppy/companion-client/src/RequestClient.js

```diff
@@ -43,8 +45,15 @@ export default class RequestClient {
     this.onReceiveResponse = this.onReceiveResponse.bind(this)
     this.allowedHeaders = ['accept', 'content-type', 'uppy-auth-token']
     this.preflightDone = false
+    this.#companionHeaders = opts !== undefined && opts.companionHeaders !== undefined ? opts.companionHeaders : {}
```

Comment:
```suggestion
    this.#companionHeaders = opts?.companionHeaders ?? {}
```
or even (I don't think it's a problem if it stays undefined)
```suggestion
    this.#companionHeaders = opts?.companionHeaders
```